apps/prairielearn/src/pages/course_instance/lookup/lookupByEnrollmentCode.ts

```diff
@@ -18,7 +22,34 @@ router.get(
     const { code, course_instance_id: courseInstanceId } = LookupCodeSchema.parse(req.query);
 
     // Look up the course instance by enrollment code
-    const courseInstance = await selectOptionalCourseInstanceByEnrollmentCode(code);
+    let courseInstance: CourseInstance | null = null;
+    try {
+      const { authzData } = await buildAuthzData({
+        authn_user: res.locals.authn_user,
+        course_id: null, // Inferred via course_instance_id
+        course_instance_id: courseInstanceId ?? null,
+        is_administrator: res.locals.is_administrator,
+        ip: req.ip ?? null,
+        req_date: res.locals.req_date,
+      });
+      if (authzData === null) {
+        throw new HttpStatusError(403, 'Access denied');
+      }
+      courseInstance = await selectOptionalCourseInstanceByEnrollmentCode({
+        enrollment_code: code,
+        requestedRole: 'Student',
+        authzData,
+      });
+    } catch (error) {
+      if (error instanceof HttpStatusError) {
```

Comment:
This is a little odd -- we want to display a more friendly error, so we catch this and use a JSON payload instead.